### DISCUSSION TITLE
possible to just execute code blocks without rendering?

### DISCUSSION THREAD

**Discussion by rhaynes74** [quarto-dev/quarto-cli#9495] [Q&A]
### Description

Occasionally I would like to use my .qmd file with python code blocks at the command line, and simply just execute the code blocks and have output to the command line, without rendering the document.  Is that possible?

**Comment by mcanouil** [comment]
See <https://quarto.org/docs/computations/render-scripts.html>.
You can also work with a Jupyter Notebook and run it via `jupyter nbconvert --execute mynotebook.ipynb`. Note that you have `quarto convert file.qmd` to get a Jupyter Notebook from a Quarto document.

**Comment by rhaynes74** [comment]
Thanks for the reply and pointer.  So I could write my file as a .py file according to the syntax described at the url you provided, and then render it as needed. 

 

Is there a way to convert a .qmd file to such a .py file? 

 

Sincerely,

R. Haynes

 

Ronald D. Haynes

Professor, Department of Mathematics & Statistics

Chair, Scientific Computing Graduate Programs

President-Elect, Canadian Applied and Industrial Mathematics Society

Memorial University of Newfoundland

 

 

 

From: Mickaël Canouil ***@***.***>
Reply-To: quarto-dev/quarto-cli ***@***.***>
Date: Friday, April 26, 2024 at 12:36 PM
To: quarto-dev/quarto-cli ***@***.***>
Cc: rhaynes74 ***@***.***>, Author ***@***.***>
Subject: Re: [quarto-dev/quarto-cli] possible to just execute code blocks without rendering? (Discussion #9495)

 

See https://quarto.org/docs/computations/render-scripts.html.
You can also work with a Jupyter Notebook and run it via jupyter nbconvert --execute mynotebook.ipynb. Note that you have quarto convert file.qmd to get a Jupyter Notebook from a Quarto document.

**Reply by mcanouil** [reply]
`quarto convert` only does Qmd <> Jupyter Notebook.

PS: when replying by emails, please do not include previous messages.
